hiss/volunteer/views.py

```diff
@@ -0,0 +1,194 @@
+        400 status code. if successful, returns a response with status 200.
+        """
+        user_email = request.data.get("email", None)
+        meal = request.data.get("meal", None)
+        restrictions = request.data.get("restrictions", None)
+
+        # Ensure that all required parameters are present
+        if not (user_email and meal and restrictions):
+            return response.Response(status=status.HTTP_400_BAD_REQUEST)
+
+        application: Application = get_object_or_404(
+            Application, user__email=user_email
+        )
+
+        # Ensure that user has checked in
+        if not application.status == STATUS_CHECKED_IN:
+            return response.Response(
+                data={"error": USER_NOT_CHECKED_IN_MSG},
+                status=status.HTTP_412_PRECONDITION_FAILED,
+            )
+
+        FoodEvent.objects.create(
+            user=application.user, meal=meal, restrictions=restrictions
+        )
+        return response.Response(status=status.HTTP_200_OK)
+
+
+class CreateWorkshopEventView(views.APIView):
+    permission_classes = [
+        permissions.IsAuthenticated & (IsVolunteer | permissions.IsAdminUser)
+    ]
+    authentication_classes = [authentication.TokenAuthentication]
+
+    def post(self, request: Request, format: str = None):
+        """
+        Creates a new WorkshopEvent (indicating that a user has attended a workshop). If the request is malformed (
+        i.e. missing the user's email), returns a Django Rest Framework Response with a 400 status code. if
+        successful, returns a response with status 200.
+        """
+        user_email = request.data.get("email", None)
+
+        # Ensure that all required parameters are present
+        if not user_email:
+            return response.Response(status=status.HTTP_400_BAD_REQUEST)
+
+        application: Application = get_object_or_404(
+            Application, user__email=user_email
+        )
+
+        # Ensure that user has checked in
+        if not application.status == STATUS_CHECKED_IN:
+            return response.Response(
+                data={"error": USER_NOT_CHECKED_IN_MSG},
+                status=status.HTTP_412_PRECONDITION_FAILED,
+            )
+
+        WorkshopEvent.objects.create(user=application.user)
+        return response.Response(status=status.HTTP_200_OK)
+
+
+class SearchView(views.APIView):
+    permission_classes = [
+        permissions.IsAuthenticated & (IsVolunteer | permissions.IsAdminUser)
+    ]
+    authentication_classes = [authentication.TokenAuthentication]
+
+    def get(self, request: Request, *args, **kwargs):
+        """
+        Performs a simple regex search for a matching application based on the user's first and last name. Creates a
+        new temporary column called "full_name" which is just "<FIRST_NAME> <LAST_NAME>", and then regex-searches the
+        query against the column, and returns all matches.
+        """
+        query = request.GET.get("q")
```

Comment:
Why do we get "q"? Does this just return all users?

Reply:
This line retrieves the value of the query parameter called `q` (which is the user's search query). [A StackOverflow answer that gave me this approach](https://stackoverflow.com/a/150518).

GET is a Python `dict`, which [offers the `.get` function for retrieving items](https://python-reference.readthedocs.io/en/latest/docs/dict/get.html)

Reply:
Ohhh I see, that makes sense